core/tests/test-dependencies/route-to-module-mapping-generator.ts

```diff
@@ -0,0 +1,441 @@
+  }
+  return childrenProperty
+    .asKindOrThrow(ts.SyntaxKind.PropertyAssignment)
+    .getInitializerOrThrow()
+    .asKindOrThrow(ts.SyntaxKind.ArrayLiteralExpression)
+    .getElements() as ObjectLiteralExpression[];
+};
+
+/**
+ * Extends the route to module mapping with a list of route nodes.
+ */
+const extendRouteToModuleMappingWithRouteNodes = (
+  routeToModuleMapping: Map<Route, string> = new Map(),
+  routeNodes: ObjectLiteralExpression[] = [],
+  parentRoutePath?: string,
+  parentRoutingModuleFilePath?: string
+): Map<Route, string> => {
+  for (const routeNode of routeNodes) {
+    routeToModuleMapping = extendMap(
+      routeToModuleMapping,
+      convertRouteNodeToMap(
+        routeNode,
+        parentRoutePath,
+        parentRoutingModuleFilePath
+      )
+    );
+  }
+  return routeToModuleMapping;
+};
+
+/**
+ * Converts a route object AST node to a map element.
+ */
+const convertRouteNodeToMap = (
+  routeNode: ObjectLiteralExpression,
+  parentRoutePath?: string,
+  parentRoutingModuleFilePath?: string
+): Map<Route, string> => {
+  let routeToModuleMapping: Map<Route, string> = new Map();
+  const path = getPathFromRouteObjectNode(routeNode);
+  if (path === undefined) {
+    throw new Error('No path was found in the route object.');
+  }
+
+  const module = getModuleFromRouteObjectNode(routeNode);
+  const pathMatch = getPathMatchFromRouteObjectNode(routeNode);
+  const childrenRouteNodes = getChildrenFromRouteObjectNode(routeNode);
+  const routePath = parentRoutePath ? `${parentRoutePath}/${path}` : path;
+
+  if (childrenRouteNodes) {
+    routeToModuleMapping = extendRouteToModuleMappingWithRouteNodes(
+      routeToModuleMapping,
+      childrenRouteNodes,
+      routePath,
+      parentRoutingModuleFilePath
+    );
+  }
+  if (module) {
+    const containingFile = routeNode.getSourceFile().getFilePath();
+    const resolvedModule = resolveModuleRelativeToRoot(module, containingFile);
+    routeToModuleMapping.set(
+      {
+        path: routePath,
+        pathMatch: pathMatch,
+      },
+      parentRoutingModuleFilePath || resolvedModule
+    );
+    const childRouteToModuleMapping = getRouteToModuleMappingFromRoutingModule(
+      resolvedModule,
+      routePath,
+      parentRoutingModuleFilePath || resolvedModule
+    );
+    routeToModuleMapping = extendMap(
+      routeToModuleMapping,
+      childRouteToModuleMapping
+    );
+  }
+  return routeToModuleMapping;
+};
+
+/**
+ * Gets the route AST nodes from a routing module file.
+ */
+const getRouteNodesFromRoutingModuleSourceFile = (
+  routingModuleSourceFile: SourceFile
+): ObjectLiteralExpression[] | undefined => {
+  const angularModuleDecorationNode = getDecoratorNodesByTextFromSourceFile(
+    routingModuleSourceFile,
+    AngularDecorators.Module
+  );
+  if (angularModuleDecorationNode.length === 0) {
+    throw new Error(
+      `No Angular Module decoration was found in the file: ${routingModuleSourceFile.getFilePath()}.`
```

Comment:
Can you keep lines under 80 characters?

Should we run linter for this files?

Reply:
It is though I don't know why prettier isn't auto formatting this, the eslintrc rule for 80 length was removed here: https://github.com/oppia/oppia/pull/19854/files#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822.

Reply:
Done.